user_auth/views.py

```diff
@@ -0,0 +1,154 @@
+from django.contrib import messages
+from django.contrib.auth import authenticate
+from django.contrib.auth import login as log_in
+from django.contrib.auth.hashers import make_password
+from django.contrib.auth.tokens import default_token_generator
+from django.core.mail import BadHeaderError, send_mail
+from django.http import HttpRequest, HttpResponse
+from django.shortcuts import redirect, render
+from django.template.loader import render_to_string
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_decode, urlsafe_base64_encode
+from django.utils.safestring import mark_safe
+
+from .forms import (
+    NewPasswordForm,
+    ResetPasswordForm,
+    UserLoginForm,
+    UserSignUpForm,
+)
+from .models import User
+
+
+def signup(request: HttpRequest):
+    if request.user.is_authenticated:
+        return redirect("/main/")
+    if request.method == "POST":
+        form = UserSignUpForm(request.POST)
+        if form.is_valid():
+            user = form.save(commit=False)
+            user.password = make_password(form.cleaned_data["password"])
+            user.save()
+            messages.success(
+                request,
+                mark_safe("Account Created: <a href='/login'>Login?</a>"),
+            )
+            return redirect("/")
+        else:
+            for errorKind, contents in form.errors.as_data().items():
+                msg = errorKind.capitalize() + ":"
+                for content in contents:
+                    msg += content.message + "\n"
+                messages.warning(request, msg)
+    form = UserSignUpForm()
+    context = {"title": "Sign Up", "form": form}
+    return render(request, "signup.html", context)
+
+
+def login(request: HttpRequest):
+    if request.user.is_authenticated:
+        return redirect("/main/")
+    if request.method == "POST":
+        form = UserLoginForm(request.POST)
+        if form.is_valid():
+            umail = form.cleaned_data["email"]
+            upasswd = form.cleaned_data["password"]
+            user = authenticate(request=request, email=umail, password=upasswd)
+            if user is not None:
+                log_in(request, user)
+                return redirect("/main/")
+            else:
+                messages.warning(request, "email/password are incorrect")
+    form = UserLoginForm()
+    context = {"title": "Login", "form": form}
+    return render(request, "login.html", context)
+
+
+def forgot_password(request: HttpRequest):
+    if request.user.is_authenticated:
+        return redirect("/main/")
+    if request.method == "POST":
```

Comment:
always make if statements shorter ...
do something like
    if not request.method == "POST":
return form.

the rest will not be inside an if statement,
if statements add code complexity